lib/mineflayer.js

```diff
@@ -14,78 +17,74 @@ module.exports = (bot, { viewDistance = 6, firstPerson = false, port = 3000, pre
+    bot.viewer.drawBoxGrid = (id, start, end, color = 'aqua') => {
+      primitives[id] = { type: 'boxgrid', start, end, color }
+      for (const socket of sockets) {
+        socket.emit('primitive', primitives[id])
+      }
     }
-  }
 
-  bot.viewer.drawPoints = (id, points, color = 0xff0000, size = 5) => {
-    primitives[id] = { type: 'points', id, points, color, size }
-    for (const socket of sockets) {
-      socket.emit('primitive', primitives[id])
+    bot.viewer.drawLine = (id, points, color = 0xff0000) => {
+      primitives[id] = { type: 'line', id, points, color }
+      for (const socket of sockets) {
+        socket.emit('primitive', primitives[id])
+      }
     }
-  }
 
-  io.on('connection', (socket) => {
-    socket.emit('version', bot.version)
-    sockets.push(socket)
+    io.on('connection', (socket) => {
+      socket.emit('version', bot.version)
+      sockets.push(socket)
 
-    const worldView = new WorldView(bot.world, viewDistance, bot.entity.position, socket)
-    worldView.init(bot.entity.position)
+      const worldView = new WorldView(
+        bot.world,
+        viewDistance,
+        bot.entity.position,
+        socket
+      )
+      worldView.init(bot.entity.position)
 
-    worldView.on('blockClicked', (block, face, button) => {
-      bot.viewer.emit('blockClicked', block, face, button)
-    })
+      worldView.on('blockClicked', (block, face, button) => {
+        bot.viewer.emit('blockClicked', block, face, button)
+      })
 
-    for (const id in primitives) {
-      socket.emit('primitive', primitives[id])
-    }
+      for (const id in primitives) {
+        socket.emit('primitive', primitives[id])
+      }
 
-    function botPosition () {
-      const packet = { pos: bot.entity.position, yaw: bot.entity.yaw, addMesh: true }
-      if (firstPerson) {
-        packet.pitch = bot.entity.pitch
+      function botPosition () {
+        const packet = { pos: bot.entity.position, addMesh: true }
+        if (firstPerson) {
+          packet.yaw = bot.entity.yaw
+          packet.pitch = bot.entity.pitch
+        }
+        socket.emit('position', packet)
+        worldView.updatePosition(bot.entity.position)
       }
-      socket.emit('position', packet)
-      worldView.updatePosition(bot.entity.position)
-    }
 
-    bot.on('move', botPosition)
-    worldView.listenToBot(bot)
-    socket.on('disconnect', () => {
-      bot.removeListener('move', botPosition)
-      worldView.removeListenersFromBot(bot)
-      sockets.splice(sockets.indexOf(socket), 1)
+      bot.on('move', botPosition)
+      worldView.listenToBot(bot)
+      socket.on('disconnect', () => {
+        bot.removeListener('move', botPosition)
+        worldView.removeListenersFromBot(bot)
+        sockets.splice(sockets.indexOf(socket), 1)
+      })
     })
-  })
+  }
+  initialize(bot)
 
   http.listen(port, () => {
     console.log(`Prismarine viewer web server running on *:${port}`)
   })
 
-  bot.viewer.close = () => {
```

Comment:
why did you remove this ?

Reply:
Sorry for not responding for a while, as I said, this is just a rebase of that pull.
I'll add it back in